internal/templating/backups/template_prebackuppod.go

```diff
@@ -0,0 +1,426 @@
+				// of the current build process
+				restoreResult := append(separator[:], pbpBytes[:]...)
+				result = append(result, restoreResult[:]...)
+			}
+		}
+	}
+	return result, nil
+}
+
+// helper function to remove the creationtimestamp from the prebackuppod pod spec so that kubectl will apply without validation errors
+func RemoveYAML(a []byte) ([]byte, error) {
+	tmpMap := map[string]interface{}{}
+	yaml.Unmarshal(a, &tmpMap)
+	if _, ok := tmpMap["spec"].(map[string]interface{})["pod"].(map[string]interface{})["metadata"].(map[string]interface{})["creationTimestamp"]; ok {
+		delete(tmpMap["spec"].(map[string]interface{})["pod"].(map[string]interface{})["metadata"].(map[string]interface{}), "creationTimestamp")
+		b, _ := yaml.Marshal(tmpMap)
+		return b, nil
+	}
+	return a, nil
+}
+
+var funcMap = template.FuncMap{
+	"VarFix": varFix,
+}
+
+// varfix just uppercases and replaces - with _ for variable names
+func varFix(s string) string {
+	return fmt.Sprintf("%s", strings.ToUpper(strings.Replace(s, "-", "_", -1)))
+}
+
+// this is just the first run at doing this, once the service template generator is introduced, this will need to be re-evaluated
+type PreBackupPods map[string]string
+
+// this is just the first run at doing this, once the service template generator is introduced, this will need to be re-evaluated
+var preBackupPodSpecs = PreBackupPods{
+	"mariadb-dbaas": `backupCommand: >-
+  /bin/sh -c "if [ ! -z $BACKUP_DB_READREPLICAS ]; then
+  BACKUP_DB_HOST=$(echo $BACKUP_DB_READREPLICAS | cut -d ',' -f1)
+  fi && dump=$(mktemp) && mysqldump --max-allowed-packet=500M --events
+  --routines --quick --add-locks --no-autocommit --single-transaction
+  --no-create-db --no-data -h $BACKUP_DB_HOST -u $BACKUP_DB_USERNAME
+  -p$BACKUP_DB_PASSWORD $BACKUP_DB_DATABASE > $dump && mysqldump
+  --max-allowed-packet=500M --events --routines --quick --add-locks
+  --no-autocommit --single-transaction --no-create-db
+  --ignore-table=$BACKUP_DB_DATABASE.watchdog --no-create-info -h
+  $BACKUP_DB_HOST -u $BACKUP_DB_USERNAME -p$BACKUP_DB_PASSWORD
+  $BACKUP_DB_DATABASE >> $dump && cat $dump && rm $dump"
+fileExtension: .{{ .Name }}.sql
+pod:
+  spec:
+    containers:
+    - args:
+      - sleep
+      - infinity
+      env:
+      - name: BACKUP_DB_HOST
+        valueFrom:
+          configMapKeyRef:
+            key: {{ .Name | VarFix }}_HOST
+            name: lagoon-env
+      - name: BACKUP_DB_USERNAME
+        valueFrom:
+          configMapKeyRef:
+            key: {{ .Name | VarFix }}_USERNAME
+            name: lagoon-env
+      - name: BACKUP_DB_PASSWORD
+        valueFrom:
+          configMapKeyRef:
+            key: {{ .Name | VarFix }}_PASSWORD
+            name: lagoon-env
+      - name: BACKUP_DB_DATABASE
+        valueFrom:
+          configMapKeyRef:
+            key: {{ .Name | VarFix }}_DATABASE
+            name: lagoon-env
+      image: imagecache.amazeeio.cloud/amazeeio/alpine-mysql-client
```

Comment:
podspec is missing `BACKUP_DB_READREPLICAS` var introduced in https://github.com/uselagoon/build-deploy-tool/pull/184

Reply:
https://github.com/uselagoon/build-deploy-tool/pull/125/files#diff-d2bb5cf6b1385ed8e8b7a4b1e08c7333394c71ca58d53895ef410f7b909f10aeR90 because read replicas are optional in the dbaas, the switch for this happens in code